### PR TITLE
Cache configuration improvement

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Allow getting cache configuration from 'named_collection' section in config or from sql created named collection.

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 6bcc7c04817e23d2b8be8f8582fb09adf906345d with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/56541/6bcc7c04817e23d2b8be8f8582fb09adf906345d/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6815351252">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/56541/6bcc7c04817e23d2b8be8f8582fb09adf906345d/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Interpreters/Cache/FileCacheSettings.cpp [hunk 1]
```diff
@@ -54,4 +50,41 @@ void FileCacheSettings::loadFromConfig(const Poco::Util::AbstractConfiguration &
...
+        bypass_cache_threshold = parseWithSizeSuffix<uint64_t>(collection.get<String>("bypass_cache_threshold"));
+
+    if (collection.has("boundary_alignment"))
+        boundary_alignment = parseWithSizeSuffix<uint64_t>(collection.get<String>("boundary_alignment"));
+
+    if (collection.has("background_download_threads"))
+        background_download_threads = collection.get<UInt64>("background_download_threads");
+
+    if (collection.has("load_metadata_threads"))
+        load_metadata_threads = collection.get<UInt64>("load_metadata_threads");
```
Comment (antonio2368) [inline src/Interpreters/Cache/FileCacheSettings.cpp:87]: why can't these be `collection.getOrDefault`?

Comment (kssenii) [reply src/Interpreters/Cache/FileCacheSettings.cpp:87]: just `FileCacheSettings` has already assigned defaults and adding `getOrDefault` here forces to write the defaults in this file again, but I thought - why do that if the defaults are already assigned? so did this way.

Comment (antonio2368) [reply src/Interpreters/Cache/FileCacheSettings.cpp:87]: I understand that, but what about `cache_hits_threshold` then?

Comment (kssenii) [reply src/Interpreters/Cache/FileCacheSettings.cpp:87]: oops, my excuse - this setting was not added by me 😅
will change it for consistency

Comment (antonio2368) [reply src/Interpreters/Cache/FileCacheSettings.cpp:87]: consistency is all I wanted 😄

**Review by antonio2368 — APPROVED** [review]

**Comment by kssenii** [comment]
> Integration tests (release) [4/4] — fail: 1, passed: 648

test_delayed_replica_failover/test.py::test

> Stateless tests (debug, s3 storage) [3/6] — fail: 1, passed: 936, skipped: 18

01280_ttl_where_group_by

> Stateless tests (release, analyzer) — fail: 1, passed: 5817, skipped: 13

02404_memory_bound_merging

**Pull request merged**